File: app/src/main/java/de/hype/hypenotify/app/core/Intent.java
package de.hype.hypenotify.app.core;

import de.hype.hypenotify.MainActivity;
import de.hype.hypenotify.app.core.interfaces.Core;

public interface Intent {
    void handleIntentInternal(android.content.Intent intent, Core core, MainActivity context);

    String intentId();
}

File: app/src/main/java/de/hype/hypenotify/app/core/Intent.kt
package de.hype.hypenotify.app.core

import android.content.Intent
import de.hype.hypenotify.app.MainActivity
import de.hype.hypenotify.app.core.interfaces.Core

interface Intent {
    fun handleIntentInternal(intent: Intent?, core: Core?, context: MainActivity?)

    fun intentId(): String?
}
